packages/react/src/components/TextInput/PasswordInput.tsx

```diff
@@ -329,18 +349,18 @@ const PasswordInput = React.forwardRef(function PasswordInput(
         data-toggle-password-visibility={inputType === 'password'}
       />
       {isFluid && <hr className={`${prefix}--text-input__divider`} />}
-      <button
-        type="button"
-        className={passwordVisibilityToggleClasses}
-        disabled={disabled}
-        onClick={handleTogglePasswordVisibility}>
-        {!disabled && (
-          <span className={`${prefix}--assistive-text`}>
-            {passwordIsVisible ? hidePasswordLabel : showPasswordLabel}
-          </span>
-        )}
-        {passwordVisibilityIcon}
-      </button>
+      <Tooltip
+        align={align}
+        className={`${prefix}--toggle-password-tooltip`}
+        label={passwordIsVisible ? hidePasswordLabel : showPasswordLabel}>
+        <button
+          type="button"
+          className={passwordVisibilityToggleClasses}
+          disabled={disabled}
```

Comment:
I think `Tooltip` handles this, but did you confirm the tooltip doesn't show if the button is disabled? I can't configure the storybook to disable that button.

Reply:
Yes, I checked the disabled functionality, it's not showing the tooltip if we pass disabled as true. I tried to remove this as well to check if tooltip is handling it or not, but then it was showing tooltip even disabled was true. So I think we have to pass it here.